backend/Models/Groups/group-model.js

```diff
@@ -0,0 +1,434 @@
+const {pool} = require('../db');
+var groupModel = {
+   getAllGroups:getAllGroups,
+   getAllGroupsOfUser:getAllGroupsOfUser,
+   addGroup:addGroup,
+   updateGroup:updateGroup,
+   deleteGroup:deleteGroup,
+   getGroupById:getGroupById,
+   addUserToGroup:addUserToGroup,
+   makeUserAdmin : makeUserAdmin,
+   customGroup : customGroup,
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {String} param0.name
+ * @param {String} param0.description
+ * @param {String} param0.confidential
+ * @return {Promise}
+ */
+
+function addGroup(
+  {
+  username,
+  name,
+  description,
+  confidential
+}
+) {
+return new Promise(async (resolve, reject) => {
+  pool.getConnection(function(error, connection) {
+    if (error) {
+      return reject(error);
+    }
+    connection.beginTransaction(function(error) {
+      if (error) {
+        return reject(error);
+      }
+      connection.query(
+          `INSERT INTO groups(name,description,confidential,created_by) VALUES (?,?,?,?) WHERE
+          (SELECT COUNT(username) FROM users WHERE (username=? AND admin=1) `,
+        [
+          name,
+          description,
+          confidential,
+          username,
+          username,
+        ],
+        (error, results, fields) => {
+          if (error) {
+            connection.rollback(function(error) {
+              connection.release();
+            });
+            return reject(error);
+          }
+          let xy = new Promise(function(resolve, reject) {
+              connection.query(
+                "INSERT INTO UserGroups(`username`,`group_id`,`admin`)" +
+                  "VALUES(?,SELECT id FROM groups WHERE name = ?,?)",
```

Comment:
I guess we can use https://github.com/mysqljs/mysql#getting-the-id-of-an-inserted-row and get Id from previous results instead of searching here.

Reply:
Fixed